File: Reto # -5/Python/PorReferencia.py
## DeathabtO

def cambiar_referencia(a, b):
    a[0], b[0] = b[0], a[0]
    return a, b

a_original = [45]
b_original = [34]

nuevo_a, nuevo_b = cambiar_referencia(a_original,b_original)


print("Por valor:")
print("Original: a =", a_original, "b =", b_original)
print("Nuevos: a =", nuevo_a, "b =", nuevo_b)

File: Reto # -5/Java/PorReferencia.java
//DeathbatO

public class PorReferencia {

    public static void main(String[] args) {
        // Definir variables
        int[] originalA = {5};
        int[] originalB = {10};

        // Llamar a la función e imprimir resultados
        int[] nuevosValores = intercambiarReferencia(originalA, originalB);
        int nuevoA = nuevosValores[0];
        int nuevoB = nuevosValores[1];

        System.out.println("\nPor referencia:");
        System.out.println("Original: a = " + originalA[0] + ", b = " + originalB[0]);
        System.out.println("Nuevos:   a = " + nuevoA + ", b = " + nuevoB);
    }

    public static int[] intercambiarReferencia(int[] a, int[] b) {
        // Intercambiar valores
        int temp = a[0];
        a[0] = b[0];
        b[0] = temp;
        // Retornar los nuevos valores
        return new int[]{a[0], b[0]};
    }
}